py/runtime.c

```diff
@@ -1507,13 +1507,13 @@ mp_obj_t mp_make_raise_obj(mp_obj_t o) {
         o = mp_call_function_n_kw(o, 0, 0, NULL);
     }
 
-    if (mp_obj_is_exception_instance(o)) {
+    mp_obj_instance_t *obj = MP_OBJ_TO_PTR(o);
+    if (mp_obj_is_exception_instance(o) && obj->subobj[0] != &mp_native_base_init_wrapper_obj) {
```

Comment:
Do you think it's safe to dereference `obj->subobj[0]` even if `obj` may not be an `mp_obj_instance_t *`?  (The latter will only be true if `mp_obj_is_instance_type(o)`.)